programs/util.h

```diff
@@ -356,9 +356,15 @@ UTIL_STATIC const char** UTIL_createFileList(const char **inputNames, unsigned i
             size_t len = strlen(inputNames[i]);
             if (buf + pos + len >= bufend) {
                 ptrdiff_t newListSize = (bufend - buf) + LIST_SIZE_INCREASE;
-                buf = (char*)realloc(buf, newListSize);
-                bufend = buf + newListSize;
-                if (!buf) return NULL;
+                buftmp = (char*)realloc(buf, newListSize);
```

Comment:
prefer declaring variable in narrowest scope.
buftmp should be declared here, instead of line 347.
It can be made `const` too.
It also does not need to be `char*` since it's not accessed anyway.
So I would lean for : 
`void* const buftmp = realloc(buf, newListSize);`
and corresponding cast at line 361